modules/perception/onboard/component/fusion_camera_detection_component.cc

```diff
@@ -181,24 +193,8 @@ bool FusionCameraDetectionComponent::Init() {
   camera_debug_writer_ =
       node_->CreateWriter<apollo::perception::camera::CameraDebug>(
           camera_debug_channel_name_);
-  if (InitSensorInfo() != cyber::SUCC) {
-    AERROR << "InitSensorInfo() failed.";
-    return false;
-  }
-  if (InitAlgorithmPlugin() != cyber::SUCC) {
-    AERROR << "InitAlgorithmPlugin() failed.";
-    return false;
-  }
-  if (InitCameraFrames() != cyber::SUCC) {
-    AERROR << "InitCameraFrames() failed.";
-    return false;
-  }
-  if (InitProjectMatrix() != cyber::SUCC) {
-    AERROR << "InitProjectMatrix() failed.";
-    return false;
-  }
-  if (InitCameraListeners() != cyber::SUCC) {
-    AERROR << "InitCameraListeners() failed.";
+  if (!EXEC_ALL_FUNS(FusionCameraDetectionComponent, this,
+    FusionCameraDetectionComponent::init_func_arry_)) {
```

Comment:
Still, personally I prefer current implementation which is more readable and intuitive, while for the new code people need to check 3 places to understand what would happen.

Added perception folks to decide.